pkg/asset/installconfig/gcp/dns.go

```diff
@@ -48,3 +59,65 @@ func GetPublicZone(ctx context.Context, project, baseDomain string) (managedZone
 	}
 	return res, nil
 }
+
+// GetBaseDomain returns a base domain chosen from among the project's public DNS zones.
+func GetBaseDomain(project string) (string, error) {
+	ctx, cancel := context.WithTimeout(context.TODO(), 1*time.Minute)
+	defer cancel()
+
+	svc, err := getDNSService(ctx)
+	if err != nil {
+		return "", err
+	}
+
+	var publicZones []string
+	req := svc.ManagedZones.List(project).Context(ctx)
+	if err := req.Pages(ctx, func(page *dns.ManagedZonesListResponse) error {
+		for _, v := range page.ManagedZones {
+			if v.Visibility != "private" {
+				publicZones = append(publicZones, strings.TrimSuffix(v.DnsName, "."))
+			}
+		}
+		return nil
+	}); err != nil {
+		return "", err
+	}
+	if len(publicZones) == 0 {
+		return "", errors.New("no domain names found in project")
+	}
+
+	var domain string
+	if err := survey.AskOne(&survey.Select{
+		Message: "Base Domain",
+		Help:    "The base domain of the cluster. All DNS records will be sub-domains of this base and will also include the cluster name.\n\nIf you don't see you intended base-domain listed, create a new public hosted zone and rerun the installer.",
+		Options: publicZones,
+	}, &domain, func(ans interface{}) error {
+		choice := ans.(string)
+		i := sort.SearchStrings(publicZones, choice)
+		if i == len(publicZones) || publicZones[i] != choice {
+			return errors.Errorf("invalid base domain %q", choice)
+		}
+		return nil
+	}); err != nil {
+		return "", errors.Wrap(err, "failed UserInput for base domain")
+	}
+
+	return domain, nil
+}
+
+// IsForbidden checks whether a response from the GPC API was forbidden,
+// indicating that a given service account cannot access the specified project.
+func IsForbidden(err error) bool {
+	if gErr, ok := err.(*googleapi.Error); ok {
```

Comment:
looking at https://github.com/openshift/installer/blob/700b75a3e2b88b7236498e0ba00708016bbc4f52/pkg/asset/installconfig/aws/basedomain.go#L19-L20

this probably can be:
```
gErr, ok := err.(*googleapi.Error)
return ok && gErr.Code == 403
```